Source/Core/DolphinWX/FrameTools.cpp

```diff
@@ -292,6 +292,70 @@ void CFrame::OpenGeneralConfiguration(wxWindowID tab_id)
   m_main_config_dialog->SetFocus();
 }
 
+void CFrame::CycleProfile(bool forward) {
+  //load the hotkey config
+  InputConfig* hk_conf = HotkeyManagerEmu::GetConfig();
+  auto hotkey_controller = hk_conf->GetController(0);
+
+  //load the wiimote config
+  InputConfig* wm_conf = Wiimote::GetConfig();
+  std::string pname(File::GetUserPath(D_CONFIG_IDX));
+
+  //find all profiles
+  pname += "Profiles/" + wm_conf->GetProfileName();
+  std::vector<std::string> sv = Common::DoFileSearch({ pname }, { ".ini" });
+
+  //remove all profiles that do not fit the filter string
+  if (!hotkey_controller->profile_filter.empty()){
+    sv.erase(
+      std::remove_if(
+        sv.begin(), sv.end(), [filter = hotkey_controller->profile_filter](std::string s)
```

Comment:
This should likely take the `s` as a `const std::string&`, otherwise the strings are unnecessarily copied.

Reply:
Probably won't make a difference with SBO but Fixed.